test-data/unit/check-deprecated.test

```diff
@@ -413,9 +415,133 @@ a.g("x")  # E: overload def (self: __main__.A, v: builtins.str) of function __ma
 a.h(1)  # E: function __main__.A.h is deprecated: use `h2`  instead
 a.h("x")  # E: function __main__.A.h is deprecated: use `h2`  instead
 
+b = B()
+b.f(1)  # E: overload def (self: __main__.A, v: builtins.int) of function __main__.A.f is deprecated: pass `str` instead
+b.f("x")
+b.g(1)
+b.g("x")  # E: overload def (self: __main__.A, v: builtins.str) of function __main__.A.g is deprecated: pass `int` instead
+b.h(1)  # E: function __main__.A.h is deprecated: use `h2`  instead
+b.h("x")  # E: function __main__.A.h is deprecated: use `h2`  instead
+
 [builtins fixtures/tuple.pyi]
 
 
+[case testDeprecatedOverloadedClassMethods]
+# flags: --enable-error-code=deprecated
+
+from typing import Iterator, Union
+from typing_extensions import deprecated, overload
+
+class A:
+    @overload
+    @classmethod
+    @deprecated("pass `str` instead")
+    def f(self, v: int) -> None: ...
+    @overload
+    @classmethod
+    def f(self, v: str) -> None: ...
+    @classmethod
+    def f(self, v: Union[int, str]) -> None: ...
+
+    @overload
+    @classmethod
+    def g(self, v: int) -> None: ...
+    @overload
+    @classmethod
+    @deprecated("pass `int` instead")
+    def g(self, v: str) -> None: ...
+    @classmethod
+    def g(self, v: Union[int, str]) -> None: ...
+
+    @overload
+    @classmethod
+    def h(self, v: int) -> A: ...
+    @overload
+    @classmethod
+    def h(self, v: str) -> A: ...
+    @deprecated("use `h2`  instead")
+    @classmethod
+    def h(self, v: Union[int, str]) -> A: ...
+
+class B(A): ...
```

Comment:
Should we also test that `A` can have regular method overriden with `B` with deprecated case?

Reply:
I think this is not really necessary.  Unless you suggest not reporting a deprecation warning if a base class offers a corresponding not deprecated method.   This seems somehow related to #18085, which has not received a response so far.  Maybe, because it is a little confusing.  Or because the topic confused me...